stor/dx.py

```diff
@@ -144,10 +144,32 @@ def download_objects(self):  # may not need it
         raise NotImplementedError
 
     def remove(self):
-        raise NotImplementedError
+        """
+        Removes a single object.
+
+        Raises:
+            ValueError: The path is invalid.
+        """
+        if not self.resource or self._is_folder():
+            raise ValueError('DXPath must point to a data object to remove a single file')
```

Comment:
aside, the function `_is_folder()` is a little bit confusing. I think it'd be clearer to call it something like `if not self._file_like()`.

That said, given that DNAnexus does have a separate concept of a folder, I'm not sure it matters to check if it's file-like the way it did on Swift/S3 (where we really can't know).

Would it make sense to error if it's a project (saying you have to use rmtree to delete a project), and then attempt to just remove the file at the specified name regardless.

A few questions that are raised here (that we should document answers to and also write tests for):

1. what should the behavior be if the file (path or dxid) does not exist? Error? Silently ignore missing?
2. what should happen if there are multiple files that match a virtual path when attempting to remove? will an exception be raised? Where does that exception occur?

Reply:
Agreed after discussion. _is_folder() does not make sense, and removed. As for the questions:
1. If the file path does not exist, a NotFoundError is raised as before.
2. If the file path is duplicate, a DuplicateError is raised as before. No files are removed in this case.